### PR TITLE
Update linting

### PR DESCRIPTION
**Pull request by matthew-reynolds**
# Pull Request

Duplicate changes from uwreact/frc_control#75, uwreact/frc_control#76, and other incremental changes to improve linting rules and ROS conventions.

## Contribution Checklist

- [x] I have read the [contributing](https://github.com/uwreact/uwreact_robot/blob/master/CONTRIBUTING.md) guide.
- [x] I have referenced relevant issues.
- [x] I have labeled accordingly.
- [x] I have detailed my changes as much as possible.

## Change Checklist

- [ ] I have added tests.
- [x] I have added necessary documentation.
- [x] I have auto formatted using clang-format or yapf.
- [ ] I have tested my changes on real hardware.

### CODE REVIEW

**Comment by Matthew-Reynolds** [comment]
Blocked by #90, will rebase once #90 is merged

**Review by wmmc88 — DISMISSED** [review]

**Review by ghost — DISMISSED** [review]

**Comment by Matthew-Reynolds** [comment]
After digging into this some more, it seems pylint does some weird things when a module only contains a single `.py` file. When there are other source files or submodules, the suppression of `no-name-in-modules` seems to work as expected, but when there are not, the behaviour is inconsistent and seems to be related to some weird caching thing. I don't fully understand it, but for the time being, the easiest solution is to just re-disable that pylint suppression. Once the package is more fleshed out, we can re-remove the suppression.

**Comment by Matthew-Reynolds** [comment]
@michaelwm @wmmc88 Build issues resolved, ready for re-review

**Review by ghost — APPROVED** [review]

**Pull request merged**
